spec/controllers/cart_controller_decorator_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Spree::Api::V2::Storefront::CartController, type: :controller do
+  let(:store) { Spree::Store.default || create(:store, default: true) }
+  let(:user)  { create(:user, public_metadata: user_metadata) }
+  let(:order) { create(:order, user: user, store: store, public_metadata: {}) }
+
+  let(:user_metadata) { {} }
+
+  before do
+    allow(controller).to receive(:current_store).and_return(store)
+    allow(controller).to receive(:spree_current_user).and_return(user)
+    allow(controller).to receive(:spree_current_order).and_return(order)
+    allow(PromotionSwitcherService).to receive_message_chain(:new, :call)
+  end
+
+  describe 'POST #create' do
+    context 'when user has SPL metadata' do
+      let(:user_metadata) do
+        {
+          'spl_no_card' => 'SPL-123',
```

Comment:
Pozwolę sobie tu zwrócić uwagę na format karty lojalnościowej, proponuję coś w stylu:
```suggestion
          'spl_no_card' => '0123456789123',
```